doc/memos/rdm-draft-alamos-lanzieri-runtime-configuration-architecture.md

```diff
@@ -0,0 +1,447 @@
+
+Any kind of storage encryption mechanism is not in the scope of this document,
+and up to the implementation of `load` and `store` or intrinsic encryption
+functionalities in the storage.
+
+A minimal RIOT Registry setup requires at least one source SF from where
+configurations are loaded and exactly one SF destination from where
+configurations are stored. Having multiple SF sources can be useful when it's
+required to migrate the data between storage facilities (e.g to migrate all
+configurations from SF A to B, register B as source and destination and add
+A as a source).
+
+A conceptual example of a SF can be found in the Appendix.
+
+## 3.3. RIOT Registry Usage Flow
+
+### Registry Initialization
+Modules declare and register **RH** for configuration groups
+in the RIOT Registry. **Storage facilities** are registered as sources and/or
+ destinations of configurations in the RIOT Registry.
+
+<img align="center" src="./files/rdm-draft-alamos-lanzieri-runtime-configuration-architecture/registry-boot.svg" width="300" />
+
+### Get, set, apply and export configurations
+At any time, the application or the configuration manager can _retrieve_ a
+configuration value (`registry_get_value`), _set_ a configuration value
+(`registry_set_value`), _apply_ configuration changes (`registry_commit`) or
+_export_ configurations using a user-defined callback function
+(`registry_export`).
+
+Note these functions don't interact with SF, so
+configuration changes are not reflected in the non-volatile storage devices
+unless `registry_store` is called (see [Load and store configurations][#5.3 Load and store configurations])
+
+The following diagram shows the process of each function. It's assumed there
+are 2 RH registered in the RIOT Registry: a _cord_ configuration
+group with a Resource Directory Server IP Address (`rd_ip_addr`) and an
+_Application_ configuration group with a `foo` configuration parameter.
+
+<img src="./files/rdm-draft-alamos-lanzieri-runtime-configuration-architecture/registry-rh.svg" />
+
+### Load and store configurations
+At any time, the application or the configuration manager can _load_ all
+configurations from all SF sources (`registry_load` function) or
+_store_ them in the SF destination (`registry_store` function).
+
+As one could expect, `registry_load` will call the SF `load`
+handler with `registry_set_value` as callback. In the a similar way,
+`registry_store` will navigate through all RH and call their
+_export_ function with the SF _store_ handler as callback.  
+
+The following diagram shows these processes:
+
+<img src="./files/rdm-draft-alamos-lanzieri-runtime-configuration-architecture/registry-storage.svg" />
+
+# 4. Configuration manager
+The configuration manager is a pseudo-module that allows a RIOT node to be
+configured from one or more communication interfaces. Examples of these
+communication interfaces could be UART, SPI or higher layers like PPP, IPv6,
+UDP, CoAP, etc. 
```

Comment:
Seems like the configuration manager is being quite loosely and generally defined here, would be good to state that (ideally wherever you're describing the scope of the document)

Reply:
Yes, indeed. I expect this section to be expanded in follow-ups. I will state that